core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerClientHandler.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.circuitbreaker;
+
+import static java.util.Objects.requireNonNull;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.circuitbreaker.CircuitBreakerCallback;
+
+/**
+ * A handler used by a {@link CircuitBreakerClient} to integrate with a CircuitBreaker.
+ * One may extend this interface to use a custom CircuitBreaker with {@link CircuitBreakerClient}.
+ */
+@UnstableApi
+@FunctionalInterface
+public interface CircuitBreakerClientHandler<I extends Request> {
+
+    /**
+     * Creates a default {@link CircuitBreakerClientHandler} which uses the provided
+     * {@link CircuitBreaker} to handle requests.
+     */
+    static <I extends Request> CircuitBreakerClientHandler<I> of(CircuitBreaker circuitBreaker) {
+        requireNonNull(circuitBreaker, "circuitBreaker");
+        return of((ctx, req) -> circuitBreaker);
+    }
+
+    /**
+     * Creates a default {@link CircuitBreakerClientHandler} which uses the provided
+     * {@link CircuitBreakerMapping} to handle requests.
+     */
+    static <I extends Request> CircuitBreakerClientHandler<I> of(CircuitBreakerMapping mapping) {
+        return new DefaultCircuitBreakerClientHandler<>(requireNonNull(mapping, "mapping"));
+    }
+
+    /**
+     * Invoked by {@link CircuitBreakerClient} right before executing a request.
+     * In a typical implementation, users may extract the appropriate CircuitBreaker
+     * implementation using the provided {@link ClientRequestContext} and {@link I} request.
```

Comment:
Javadoc does not seem to link to `I`. How about just using `{@link Request}`?

<img width="679" alt="image" src="https://user-images.githubusercontent.com/1866157/201342910-4a992221-a177-44b9-85f9-c7a52b429b63.png">